packages/markdown/src/index.ts

```diff
@@ -64,35 +71,53 @@ export function createProcessor({
   mdastVisitors,
   hastVisitors,
   sanitize = true,
+  allowHtml = true,
 }: {
   mdastVisitors?: ((ast: MdastRoot) => undefined)[];
   hastVisitors?: ((ast: HastRoot) => undefined)[];
   sanitize?: boolean;
+  allowHtml?: boolean;
 } = {}) {
-  const htmlConversion = (mdastVisitors ?? [])
-    .reduce<Processor<MdastRoot, MdastRoot, MdastRoot | undefined>>(
-      (processor, visitor) => processor.use(makeHandler(visitor)),
-      unified().use(remarkParse).use(remarkMath),
-    )
-    .use(makeHandler(visitMathBlock))
-    .use(remarkGfm)
-    .use(remark2rehype, { allowDangerousHtml: true })
-    .use(rehypeRaw);
-  return (hastVisitors ?? [])
-    .reduce(
-      (processor, visitor) => processor.use(makeHandler(visitor)),
-      sanitize ? htmlConversion.use(rehypeSanitize) : htmlConversion,
-    )
-    .use(rehypeStringify);
+  const plugins: (Plugin<any, any, any> | PluginTuple<any, any, any>)[] = [
+    remarkParse,
+    remarkMath,
+    ...(mdastVisitors ?? []).map((visitor) => makeHandler(visitor)),
+    makeHandler(visitMathBlock),
+    remarkGfm,
+    [remark2rehype, { allowDangerousHtml: allowHtml }],
+    ...(!allowHtml ? [] : [rehypeRaw]),
+    ...(sanitize ? [rehypeSanitize] : []),
+    ...(hastVisitors ?? []).map((visitor) => makeHandler(visitor)),
+    rehypeStringify,
+  ];
+
+  return plugins.reduce((processor: Processor, plugin) => {
+    if (Array.isArray(plugin)) return processor.use(...plugin);
+    return processor.use(plugin);
+  }, unified());
 }
 
-const defaultProcessor = createProcessor();
-const inlineProcessor = createProcessor({ hastVisitors: [visitCheckSingleParagraph] });
+const processorCache = new Map<string, Processor>();
+function getProcessor(options: { inline?: boolean; allowHtml?: boolean } = {}): Processor {
+  const key = `${options.inline}:${options.allowHtml}`;
```

Comment:
We probably want `inline: undefined` and `inline: false` to be considered equivalent, right? Should this be:

```suggestion
  const key = `${options.inline ?? false}:${options.allowHtml ?? true}`;
```

Reply:
Better yet, we can just make these non-optional, as `markdownToHtml` always provides defaults.

Reply:
Made these non-optional in [0b17b42](https://github.com/PrairieLearn/PrairieLearn/pull/11953/commits/0b17b42ccf941335c4a00d1e45af733812ba1fd8).